### PR TITLE
wip: Basic Portfolio Markup

### PR DESCRIPTION
**Pull request by mindfulme**
#### What's this PR do?
Adds Portfolio table component
#### Where should the reviewer start?
From Portfolio.vue 
#### How should this be manually tested?
Go to Dashboard components `http://localhost:8080/#/`
#### Any background context you want to provide?
No
#### What are the relevant tickets?
Basic Portfolio Markup
#### Screenshots (if appropriate)
screen shot 2018-10-11 at 03 04 11
screen shot 2018-10-11 at 03 04 19

#### Questions:
How to pass data to the table? Via props?

#### Url to the deployment:
http://staging-ui.trusty.apasia.tech:8080/portfolio/#/

### CODE REVIEW

File: src/components/Portfolio/Portfolio.vue [hunk 1]
```diff
@@ -0,0 +1,83 @@
+<template>
+  <div class="portfolio">
+    <table>
```
Comment (roma219) [inline src/components/Portfolio/Portfolio.vue]: No tables - use grid instead, css or tailwind has capabilities - https://tailwindcss.com/docs/examples/grids

File: src/components/Portfolio/Portfolio.vue [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+<style scoped lang="scss">
+
+.portfolio {
+  font-family: config("fonts.gotham");
+}
+
+table {
+  width: 100%;
+  border-right: 10px solid config('colors.table-bg');
```
Comment (roma219) [inline src/components/Portfolio/Portfolio.vue]: tables-bg ? it is not background

File: src/components/Portfolio/Portfolio.vue [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  border-left: 10px solid config('colors.table-bg');
+  border-bottom: 10px solid config('colors.table-bg');
```
Comment (roma219) [inline src/components/Portfolio/Portfolio.vue]: 30px border?

File: src/views/Dashboard.vue [hunk 4]
```diff
@@ -1,10 +1,31 @@
 <template>
-  <div class="dashboard">dashboard!</div>
+  <div class="dashboard">
+    dashboard!
+    <Card
+      :title="'portfolio'">
+      <div slot="header">BALANCE 6754 USD</div>
+      <div slot="body">
+        <Tabs :tabs="['SHOW ALL ASSETS', 'SHOW BALANCES']">
```
Comment (roma219) [inline src/views/Dashboard.vue]: No need for tabs

**Review by roma219 — CHANGES_REQUESTED** [review]
Don't use `<table>`. Use grid instead. Portfolio is a view and should be in a View folder. You should have `Portfolio`, `PortfolioHeader`, `PortfolioItem` components.

**Comment by roma219** [comment]
Also you removed some part with the url from the PR message.

> #### URL
> Provide a link to your deploy from http://staging-ui.trusty.apasia.tech:8080/

**Comment by roma219** [comment]
Also read https://github.com/bitshares/bitshares-community-ui/blob/staging/docs/dev-guide.md and fix this branch/pr

**Comment by mindfulme** [comment]
screen shot 2018-10-12 at 18 32 30

@roma219 quick review, please

File: .gitignore [hunk 5]
```diff
@@ -1,7 +1,7 @@
 .DS_Store
 node_modules
 /dist
-
+Procfile
```
Comment (roma219) [inline .gitignore]: what is this?

File: src/views/Portfolio/Portfolio.vue [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  display: grid;
+  grid-template-columns: repeat(4, 1fr);
+}
+
+.grid > span {
+    padding: 8px 4px;
+    font-size: config('textSizes.sm');
+}
+
+.grid > span > strong {
```
Comment (roma219) [inline src/views/Portfolio/Portfolio.vue]: Better use semantic class selectors instead
`.portfolio-table .header-item`

File: src/views/Portfolio/PortfolioHeader.vue [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+<style scoped lang="scss">
+.portfolio {
+  font-family: config("fonts.gotham");
+  font-size: config('textSizes.sm');
+  display: flex;
+  color: config('colors.text-primary');
+}
+
+.textWeak {
```
Comment (roma219) [inline src/views/Portfolio/PortfolioHeader.vue]: .text-weak

File: src/views/Portfolio/PortfolioHeader.vue [hunk 8]
```diff
@@ -0,0 +1,26 @@
+<template>
+  <div class="portfolio">
```
Comment (roma219) [inline src/views/Portfolio/PortfolioHeader.vue]: Don't make class duplications, name is semanticly, like 'portfoli-header`

**Review by roma219 — CHANGES_REQUESTED** [review]
See comments. The Data will be an array of objects, similar to this: 
```
[{
 name: 'BTC',
 token: 0.345,
 fiatValue: 2345,
percent: 12
}, {
...
}]
```
You should render `PortfolioItem` components inside `Portfolio` interating over this array with `v-for`

**Comment by roma219** [comment]
Also:
image
Provide a link, rename branch and PR

**Comment by mindfulme** [comment]
screen shot 2018-10-13 at 00 11 16

http://staging-ui.trusty.apasia.tech:8080/portfolio/#/

**Comment by mindfulme** [comment]
@roma219 I only didn't rename the branch name as it causes the PR to be closed automatically, everything else should be fine

**Comment by roma219** [comment]
@MindfulMe 
image
Should be aligned

Also the table should take all 100% width

File: Procfile [hunk 9]
```diff
@@ -0,0 +1 @@
+Web: 'rpm run serve'
```
Comment (roma219) [inline Procfile]: what is this?

File: src/views/Dashboard.vue [hunk 10]
```diff
@@ -1,40 +1,62 @@
...
+      <Card
+        :title="'portfolio'"
+        class="lg:w-1/3"
+      >
+        <div slot="header">
+          <PortfolioHeader />
+        </div>
+        <div slot="body">
+          <Portfolio>
+            <div slot="item">
```
Comment (roma219) [inline src/views/Dashboard.vue]: You don't need slot in Portfolio, no components are gonna be placed like that inside portfolio
PortfolioItem should represent 1 row that is rendered via `v-for` inside `Portfolio`

File: src/views/Portfolio/PortfolioItem.vue [hunk 11]
```diff
@@ -0,0 +1,109 @@
+<template>
+  <div>
+    <div
+      v-for="(item, index) in items"
```
Comment (roma219) [inline src/views/Portfolio/PortfolioItem.vue]: PortfolioItem should be just 1 row and the iterated over in Portfolio, like this:
`<PortfolioItem v-for="(item, index) in items" :item="item" />

File: src/views/Portfolio/PortfolioItem.vue [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+      <span>{{ item.token }}</span>
+      <span>{{ item.fiatValue }}</span>
+      <span>{{ item.percent }}</span>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  data() {
```
Comment (roma219) [inline src/views/Portfolio/PortfolioItem.vue]: This data should be in Portfolio and send to each PortfolioItem via props

File: src/views/Portfolio/PortfolioItem.vue [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+  grid-template-columns: repeat(4, 1fr);
+  line-height: 5px;
+}
+
+.grid-items span {
+    padding: 8px 15px;
+    font-size: config('textSizes.sm');
+}
+
+@media screen and (max-width: 990px) {
```
Comment (roma219) [inline src/views/Portfolio/PortfolioItem.vue]: No need for media queries for now, at this point we only need a basic table. Also, see https://tailwindcss.com/docs/responsive-design

**Review by roma219 — CHANGES_REQUESTED** [review]
@MindfulMe This task needs to be done asap

**Comment by mindfulme** [comment]
okay, got it

**Comment by mindfulme** [comment]
@roma219 done, now I guess I should review everything once more

**Comment by mindfulme** [comment]
reviewed - now must be ok, responsive with the iteration as needed

File: src/views/Portfolio/Portfolio.vue [hunk 14]
```diff
@@ -28,7 +28,13 @@
       <span><strong>24h%</strong></span>
       <span><strong>7d%</strong></span>
     </div>
-    <PortfolioItem :items="items" />
+    <div
```
Comment (roma219) [inline src/views/Portfolio/Portfolio.vue:31]: No need for this div, you can iterate over PortfolioItem the same way aka:
`<PortfolioItem v-for ...etc />

**Review by roma219 — CHANGES_REQUESTED** [review]
small fix

**Comment by roma219** [comment]
@MindfulMe This should be aligned the same distance from the left like Portoflio title
image

Also the table should take all 100% width

**Comment by mindfulme** [comment]
Got it, will apply something better

**Comment by mindfulme** [comment]
removed the horizontal padding

**Comment by roma219** [comment]
@MindfulMe Fix MR conflict

File: src/views/Dashboard.vue [hunk 15]
```diff
@@ -1,49 +1,59 @@
...
-        class="uppercase">
-        under construction
-      </div>
-    </Card>
-
-    <Card
-      class="lg:w-1/3 scrollable"
-      title="Transactions">
-      <div slot="body">
-        <TransactionsHistory/>
```
Comment (roma219) [inline src/views/Dashboard.vue:34]: @MindfulMe  Why did you remove this card with Transactions widget? Fix please

**Review by roma219 — CHANGES_REQUESTED** [review]
Return transactions widget and just place you Portfolio components into "Account" widget

**Comment by roma219** [comment]
@MindfulMe Did you actually run the app after merge? There are errors, build is failing:
image

**Comment by roma219** [comment]
@MindfulMe fix asap

**Comment by mindfulme** [comment]
almost done

**Comment by mindfulme** [comment]
@roma219 works now

**Review by roma219 — APPROVED** [review]

**Pull request merged**